worker-store-fs/src/main/java/com/hpe/caf/worker/datastore/fs/FileSystemDataStore.java

```diff
@@ -79,7 +79,28 @@ public FileSystemDataStore(final FileSystemDataStoreConfiguration config)
 
         LOG.debug("Initialised");
     }
-
+    
+    private boolean doesFileExist(final Path path)
+    {
+        try
+        {
+            path.getFileSystem().provider().checkAccess(path);
+        }
+        catch(final IOException ex)
+        {
+            if(ex instanceof NoSuchFileException)
+            {
+                LOG.error("file does not exist", ex);
+            }
+            else
+            {
+                LOG.error("File existence check failed ", ex);
+            }
+            return false;
+        }
+        return true;
+    }
```

Comment:
As as aside, this function would normally be formatted like this:

    private boolean doesFileExist(final Path path)
    {
        try {
            path.getFileSystem().provider().checkAccess(path);
        } catch (final IOException ex) {
            if (ex instanceof NoSuchFileException) {
                LOG.error("file does not exist", ex);
            } else {
                LOG.error("File existence check failed ", ex);
            }
            return false;
        }
        return true;
    }